### ISSUE TITLE
`The SDK 'Uno.Sdk' specified could not be found.` when adding a Library project without global.json

### ISSUE THREAD

**Issue by AndrewKeepCoding** [unoplatform/uno#18619]
### Current behavior

I get the following error dialog when creating/adding a new Uno Platform Library project.

![image](https://github.com/user-attachments/assets/96d4868e-ccc4-4928-9c39-d517c052d1fb)

These files are created but not loaded to the solution.

![image](https://github.com/user-attachments/assets/2b9c59c1-0d53-4cd9-8618-d9d38963c485)

The created **global.json** and **.csproj** look good to me.

**global.json**
```json
{
  "msbuild-sdks": {
    "Uno.Sdk": "5.4.10"
  }
}
```
**.csproj**
```xml
<Project Sdk="Uno.Sdk">
  <PropertyGroup>
    <TargetFrameworks>net8.0;net8.0-ios;net8.0-maccatalyst;net8.0-android;net8.0-windows10.0.19041;net8.0-browserwasm;net8.0-desktop</TargetFrameworks>
    <UnoSingleProject>true</UnoSingleProject>
    <OutputType>Library</OutputType>
    <!-- Ensures the .xr.xml files are generated in a proper layout folder -->
    <GenerateLibraryLayout>true</GenerateLibraryLayout>
    <ImplicitUsings>enable</ImplicitUsings>
    <Nullable>enable</Nullable>

    <!--
      UnoFeatures let's you quickly add and manage implicit package references based on the features you want to use.
      https://aka.platform.uno/singleproject-features
    -->
    <!--
    <UnoFeatures></UnoFeatures>
    -->
  </PropertyGroup>

  <ItemGroup Condition="$([MSBuild]::GetTargetPlatformIdentifier('$(TargetFramework)')) == 'windows'">
    <!--
    If you encounter this error message:

      error NETSDK1148: A referenced assembly was compiled using a newer version of Microsoft.Windows.SDK.NET.dll.
      Please update to a newer .NET SDK in order to reference this assembly.

    This means that the two packages below must be aligned with the "build" version number of
    the "Microsoft.Windows.SDK.BuildTools" package above, and the "revision" version number
    must be the highest found in https://www.nuget.org/packages/Microsoft.Windows.SDK.NET.Ref.
    -->
    <!-- <FrameworkReference Update="Microsoft.Windows.SDK.NET.Ref" RuntimeFrameworkVersion="10.0.22621.28" />
    <FrameworkReference Update="Microsoft.Windows.SDK.NET.Ref" TargetingPackVersion="10.0.22621.28" /> -->
  </ItemGroup>
</Project>
```

### Expected behavior

Creates/Adds an Uno Platform Library without errors.

### Template Host

Visual Studio

### Host OS

Windows

### Template Options

Just [Add a new project] -> [Uno Platform Library] on Visual Studio 2022 [Preview].

### Template Version

```shell
Uno Platform Extension v5.4.14.16
```

### Anything else we need to know?
- No errors when creating a new Uno Platform app project.
- `uno-check` runs successfully.
- Followed these steps but didn't help.
  https://platform.uno/docs/articles/common-issues-all-ides.html#could-not-resolve-sdk-unosdk
- At least 2 users (including myself) are getting the same error.

**Comment by jeromelaban** [comment]
Thanks for the report. 

This is likely a misconfiguration of Visual Studio of some kind. Can you validate that the app that has been created (it should be on the disk even if VS does not load it) can be compiled using `dotnet new` on the command line?

**Comment by AndrewKeepCoding** [comment]
Hi @jeromelaban !
Do you mean `dotnet build`? The first time got an error mentioning `WindowsSdkPackageVersion`.
After adding:
```xml
<WindowsSdkPackageVersion>10.0.19041.38</WindowsSdkPackageVersion>
```
on the Uno Library *.csproj, the build successed.

**Comment by jeromelaban** [comment]
> Do you mean dotnet build?

I did, my apologies :)

It's interesting that you're getting that particular error. Is it loading in VS after this?

**Comment by AndrewKeepCoding** [comment]
>  Is it loading in VS after this?

No. Same error. I'm not the only one by the way. @ArchieCoder gets the same behavior. 🤔

**Comment by jeromelaban** [comment]
Would you have private nuget feed that is not working in your nuget feeds list?

**Comment by AndrewKeepCoding** [comment]
Just in case, here is my Visual Studio version:

Microsoft Visual Studio Enterprise 2022 (64-bit) - Preview
Version 17.12.0 Preview 4.0

**Comment by ArchieCoder** [comment]
I tried in Andrew's solution and yeah I got the bug.

I tried in my solution and it works.

I'm on regular VS version.

**Comment by jeromelaban** [comment]
@AndrewKeepCoding are you able to provide a repro?

**Comment by AndrewKeepCoding** [comment]
In my environment, I get the same behavior trying to create/add a brand-new Uno Platform Library...🤔

**Comment by jeromelaban** [comment]
Thanks. Indeed I can reproduce the problem. The issue here is caused by the fact that VS does not provide a good way for determining that global.json should be used or not, and where.

The workaround, so far, is to go to the folder that was created, move the global.json file next to the sln, then add the csproj manually.

**Comment by AndrewKeepCoding** [comment]
@jeromelaban Thank you for the quick support!
I confirmed that the workaround works!
